bpf/lib/nodeport.h

```diff
@@ -161,6 +161,37 @@ static __always_inline bool dsr_is_too_big(struct __ctx_buff *ctx __maybe_unused
 	return false;
 }
 
+static __always_inline int
+maybe_add_l2_hdr(struct __ctx_buff *ctx __maybe_unused,
+		 __u32 ifindex __maybe_unused,
+		 bool *l2_hdr_required __maybe_unused)
+{
+/* XDP does not support L2-less devices, so skip the checks. */
```

Comment:
For XDP, shouldn't these just be compiled out? (`IS_L3_DEV` always false, `ETH_HLEN` always 14)

Reply:
Let me check whether compiler removes them.

Reply:
OK, this one gets compiled out too. I'm going to drop the redundant `__ctx_is` checks.

Reply:
Sounds good, the less such checks where not needed the better.